File: app/src/main/java/com/babestudios/companieshouse/data/model/filinghistory/FilingHistoryItem.java
package com.babestudios.companieshouse.data.model.filinghistory;


import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class FilingHistoryItem {
	@SerializedName("date")
	public String date;

	@SerializedName("type")
	public String type;

	@SerializedName("links")
	public FilingHistoryLinks links;

	@SerializedName("category")
	public String category;

	@SerializedName("action_date")
	public String actionDate;

	@SerializedName("description")
	public String description;

	@SerializedName("description_values")
	public DescriptionValues descriptionValues;

	@SerializedName("pages")
	public Integer pages;

	@SerializedName("barcode")
	public String barcode;

	@SerializedName("transaction_id")
	public String transactionId;

	@SerializedName("associated_filings")
	public List<AssociatedFiling> associatedFilings = new ArrayList<AssociatedFiling>();

	@SerializedName("paper_filed")
	public Boolean paperFiled;

}

File: app/src/main/java/com/babestudios/companyinfouk/data/model/filinghistory/FilingHistoryItem.kt
package com.babestudios.companyinfouk.data.model.filinghistory


import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

import java.util.ArrayList

@Parcelize
class FilingHistoryItem(
		@SerializedName("date")
		var date: String? = null,

		@SerializedName("type")
		var type: String? = null,

		@SerializedName("links")
		var links: FilingHistoryLinks? = null,

		@SerializedName("category")
		var category: Category? = null,

		@SerializedName("subcategory")
		var subcategory: String? = null,

		@SerializedName("action_date")
		var actionDate: String? = null,

		@SerializedName("description")
		var description: String? = null,

		@SerializedName("description_values")
		var descriptionValues: DescriptionValues? = null,

		@SerializedName("pages")
		var pages: Int? = null,

		@SerializedName("barcode")
		var barcode: String? = null,

		@SerializedName("transaction_id")
		var transactionId: String? = null,

		@SerializedName("associated_filings")
		var associatedFilings: List<AssociatedFiling> = ArrayList(),

		@SerializedName("paper_filed")
		var paperFiled: Boolean? = null) : Parcelable
